### PR TITLE
Backup list

### PR DESCRIPTION
**Pull request by MiaAltieri**
## Summary
Adds backup list action **and** solves a small bug in which when calling to `resync` the `pbm_agent` doesn't recieve this command before checking if `resync` is complete. This gives a false positive that `resync` is complete, so it is necessary to wait 2 seconds before checking `resync`

### CODE REVIEW

File: lib/charms/mongodb/v0/mongodb_backups.py [hunk 1]
```diff
@@ -239,6 +243,28 @@ def _on_create_backup_action(self, event) -> None:
...
+
+        # cannot list backup if pbm is not resynced and ready
+        try:
+            self._verify_resync()
+        except ResyncError:
+            event.defer()
+            logger.debug(
+                "Sync-ing configurations needs more time, must wait before listing backups."
+            )
+            return
```
Comment (marceloneppel) [inline lib/charms/mongodb/v0/mongodb_backups.py:259]: Shouldn't you also stop the flow of execution if `_verify_resync` sets a `BlockedStatus` due to incompatible S3 settings or invalid credentials?

Comment (delgod) [reply lib/charms/mongodb/v0/mongodb_backups.py:259]: I think we should exit with an error if invalid credentials

**Review by delgod — CHANGES_REQUESTED** [review]

File: lib/charms/mongodb/v0/mongodb_backups.py (hunk 1 above)
Comment (MiaAltieri) [reply lib/charms/mongodb/v0/mongodb_backups.py:259]: gotcha

Comment (MiaAltieri) [reply lib/charms/mongodb/v0/mongodb_backups.py:259]: agreed

**Comment by MiaAltieri** [comment]
Failing unit tests will be updated [in the unit test PR](https://github.com/canonical/mongodb-operator/pull/152#issuecomment-1411890281)

File: lib/charms/mongodb/v0/mongodb_backups.py [hunk 2]
```diff
@@ -129,49 +148,47 @@ def _set_config_options(self, pbm_configs):
...
-                self.charm.unit.status = BlockedStatus("s3 credentials are incorrect.")
-                return
-
-            logger.error(e)
-            self.charm.unit.status = BlockedStatus("s3 configurations are incompatible.")
-            return
+                return BlockedStatus("s3 credentials are incorrect.")
 
-        self.charm.unit.status = ActiveStatus("")
+            return BlockedStatus("s3 configurations are incompatible.")
```
Comment (delgod) [inline lib/charms/mongodb/v0/mongodb_backups.py:184]: hm, maybe, not sure
```suggestion
            return BlockedStatus("s3 configurations are incompatible.")
        return ActiveStatus("")
```

File: src/charm.py [hunk 3]
```diff
@@ -369,8 +369,21 @@ def _on_update_status(self, event):
...
 
-        # update the units status based on it's replica set config.
-        self.unit.status = build_unit_status(self.mongodb_config, self._unit_ip(self.unit))
+        # update the units status based on it's replica set config and backup status. An error in
+        # the status of MongoDB takes precedence over pbm status.
+        mongodb_status = build_unit_status(self.mongodb_config, self._unit_ip(self.unit))
+        pbm_status = self.backups._get_pbm_status()
+        if (
+            isinstance(mongodb_status, WaitingStatus)
+            or isinstance(mongodb_status, BlockedStatus)
```
Comment (delgod) [inline src/charm.py:378]: maybe, not sure
```suggestion
            not isinstance(mongodb_status, ActiveStatus)
```

**Review by delgod — APPROVED** [review]

File: lib/charms/mongodb/v0/mongodb_backups.py (hunk 2 above)
Comment (MiaAltieri) [reply lib/charms/mongodb/v0/mongodb_backups.py:184]: I don't see the harm in this. AFAICT this is a solid suggestion. Thank you Mykola :)

File: src/charm.py (hunk 3 above)
Comment (MiaAltieri) [reply src/charm.py:378]: this is also a solid suggestion thank you Mykola

**Review by marceloneppel — APPROVED** [review]
LGTM! Great job, Mia!

**Pull request merged**